# Skewness and Kurtosis of Investment Returns

In this practice, we will observe the violation of the Normal Assumption in the historical data of Investment Returns.

I will use the data from *__EDHEC Risk Hedge Fund Indices__*

Source: https://risk.edhec.edu/all-downloads-hedge-funds-indices


In [1]:
%load_ext autoreload
%autoreload 2


In [2]:
import pandas as pd
import edhec_risk_kit_105 as erk
hfi = erk.get_hfi_returns()
hfi.head()

Convertible Arbitrage  CTA Global  Distressed Securities  \
date                                                                
1997-01                 0.0119      0.0393                 0.0178   
1997-02                 0.0123      0.0298                 0.0122   
1997-03                 0.0078     -0.0021                -0.0012   
1997-04                 0.0086     -0.0170                 0.0030   
1997-05                 0.0156     -0.0015                 0.0233   

         Emerging Markets  Equity Market Neutral  Event Driven  \
date                                                             
1997-01            0.0791                 0.0189        0.0213   
1997-02            0.0525                 0.0101        0.0084   
1997-03           -0.0120                 0.0016       -0.0023   
1997-04            0.0119                 0.0119       -0.0005   
1997-05            0.0315                 0.0189        0.0346   

         Fixed Income Arbitrage  Global Macro  Long/Short Equity  \
date                                                               
1997-01                  0.0191        0.0573             0.0281   
1997-02                  0.0122        0.0175            -0.0006   
1997-03                  0.0109       -0.0119            -0.0084   
1997-04                  0.0130        0.0172             0.0084   
1997-05                  0.0118        0.0108             0.0394   

         Merger Arbitrage  Relative Value  Short Selling  Funds Of Funds  
date                                                                      
1997-01            0.0150          0.0180        -0.0166          0.0317  
1997-02            0.0034          0.0118         0.0426          0.0106  
1997-03            0.0060          0.0010         0.0778         -0.0077  
1997-04           -0.0001          0.0122        -0.0129          0.0009  
1997-05            0.0197          0.0173        -0.0737          0.0275

# A. Skewness
## A.1 Calculate the Skewness of Investment Returns

In [3]:
List1=[hfi.mean(),hfi.median(),hfi.mean()<hfi.median()]
List2=["Mean", "Median", "Mean < Median"]
Dist_Check=pd.concat(List1, axis="columns")
Dist_Check.columns=List2
Dist_Check

Mean  Median  Mean < Median
Convertible Arbitrage   0.005508  0.0065           True
CTA Global              0.004074  0.0014          False
Distressed Securities   0.006946  0.0089           True
Emerging Markets        0.006253  0.0096           True
Equity Market Neutral   0.004498  0.0051           True
Event Driven            0.006344  0.0084           True
Fixed Income Arbitrage  0.004365  0.0055           True
Global Macro            0.005403  0.0038          False
Long/Short Equity       0.006331  0.0079           True
Merger Arbitrage        0.005356  0.0060           True
Relative Value          0.005792  0.0067           True
Short Selling          -0.001701 -0.0053          False
Funds Of Funds          0.004262  0.0052           True

In [4]:
import scipy.stats
Dist_Check["Skewness"]=scipy.stats.skew(hfi)
Dist_Check

Mean  Median  Mean < Median  Skewness
Convertible Arbitrage   0.005508  0.0065           True -2.639592
CTA Global              0.004074  0.0014          False  0.173699
Distressed Securities   0.006946  0.0089           True -1.300842
Emerging Markets        0.006253  0.0096           True -1.167067
Equity Market Neutral   0.004498  0.0051           True -2.124435
Event Driven            0.006344  0.0084           True -1.409154
Fixed Income Arbitrage  0.004365  0.0055           True -3.940320
Global Macro            0.005403  0.0038          False  0.982922
Long/Short Equity       0.006331  0.0079           True -0.390227
Merger Arbitrage        0.005356  0.0060           True -1.320083
Relative Value          0.005792  0.0067           True -1.815470
Short Selling          -0.001701 -0.0053          False  0.767975
Funds Of Funds          0.004262  0.0052           True -0.361783

## A.2 The Skewness of Normal Random Sample
Let's use numpy randome normal to populate a sample of normally distributed variable.

Let's make it comparable with the observations in 'hfi' dataframe.
In order to do so, let's find out the size of the dataframe 'hfi'.


In [5]:
hfi.shape

(263, 13)

Right!, Then, we can populate 263 of normal random samples.

In [6]:
import numpy as np
normal_returns = np.random.normal(0, .15, size=(hfi.shape[0],1))

Let's check the skewness of the freshly populated normal random samples.

In [7]:
scipy.stats.skew(normal_returns)

array([-0.37733551])

# B. Kurtosis
## B.1 Kurtosis of Investment Returns

### <span style='color: red;'> *__Caution__* </span>

Somehow, *__scipy.stats.kurtosis__* does not calculate Kurtosis.

It calculates *__Excess Kurtosis__*, which is Kurtosis beyond 3. 


## <span style='color: blue;'> *__scipy.stats.kurtosis  ==> 'Excess Kurtosis' or Kurtosis - 3__*  </span>



In [8]:
Dist_Check["Excess Kurtosis"]=scipy.stats.kurtosis(hfi)
Dist_Check

Mean  Median  Mean < Median  Skewness  \
Convertible Arbitrage   0.005508  0.0065           True -2.639592   
CTA Global              0.004074  0.0014          False  0.173699   
Distressed Securities   0.006946  0.0089           True -1.300842   
Emerging Markets        0.006253  0.0096           True -1.167067   
Equity Market Neutral   0.004498  0.0051           True -2.124435   
Event Driven            0.006344  0.0084           True -1.409154   
Fixed Income Arbitrage  0.004365  0.0055           True -3.940320   
Global Macro            0.005403  0.0038          False  0.982922   
Long/Short Equity       0.006331  0.0079           True -0.390227   
Merger Arbitrage        0.005356  0.0060           True -1.320083   
Relative Value          0.005792  0.0067           True -1.815470   
Short Selling          -0.001701 -0.0053          False  0.767975   
Funds Of Funds          0.004262  0.0052           True -0.361783   

                        Excess Kurtosis  
Convertible Arbitrage         20.280834  
CTA Global                    -0.047040  
Distressed Securities          4.889983  
Emerging Markets               6.250788  
Equity Market Neutral         14.218555  
Event Driven                   5.035828  
Fixed Income Arbitrage        26.842199  
Global Macro                   2.741679  
Long/Short Equity              1.523893  
Merger Arbitrage               5.738950  
Relative Value                 9.121208  
Short Selling                  3.117772  
Funds Of Funds                 4.070153

## B.2 Kurtosis of Normal Random Sample

### <span style='color: red;'> *__Caution__* </span>

Repeatedly, *__scipy.stats.kurtosis__* does not calculate Kurtosis.

It calculates *__Excess Kurtosis__*, which is Kurtosis beyond 3. 

So, the output of *__scipy.stats.kurtosis__* for the normal random sample would not be close to 3, instead it would be close to 0. 

In [9]:
scipy.stats.kurtosis(normal_returns)

array([0.16322147])

# C. Testing Normality: Jarque-Bera test

What can we do in order to assess the <span style='color: blue;'>*__Normality__* </span> of a distribution?

> The Jarque-Bera test tests whether the sample data has the skewness and kurtosis matching a normal distribution.

> Note that this test only works for a large enough number of data samples (>2000) as the test statistic asymptotically has a Chi-squared distribution with 2 degrees of freedom.


 * Source (scipy document): https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.jarque_bera.html
 
> *__The null hypothesis__* for the test is that the data is normally distributed; the alternate hypothesis is that the data does not come from a normal distribution.

> In general, a large J-B value indicates that errors are not normally distributed.

> Unfortunately, most statistical software does not support this test. In order to interpret results, you may need to do a little comparison (and so you should be intimately familiar with hypothesis testing). Checking p-values is always a good idea. For example, a tiny p-value and a large chi-square value from this test means that you can reject the null hypothesis that the data is normally distributed.


 * Source (Statistics): Stephanie Glen. "Jarque-Bera Test" From StatisticsHowTo.com: Elementary Statistics for the rest of us! https://www.statisticshowto.com/jarque-bera-test/



In [10]:
scipy.stats.jarque_bera(normal_returns)

(6.533025188894738, 0.03813920228592793)

In [11]:
scipy.stats.jarque_bera(normal_returns)[0]

6.533025188894738

In [12]:
scipy.stats.jarque_bera(normal_returns)[1]

0.03813920228592793

Now, we define a function to assess the normality of data, using *__Jacque Bera__*

Let's use 0.01 as the criteria for *__p value__*

In [13]:
def JB_normality_test(r, level=0.01):
    """
    Applies the Jarque-Bera test to determine if a Series is normal or not
    Test is applied at the 1% level by default
    Returns True if the hypothesis of normality is accepted, False otherwise
    """
    if isinstance(r, pd.DataFrame):
        return r.aggregate(is_normal)
    else:
        statistic, p_value = scipy.stats.jarque_bera(r)
        return p_value > level


Use *__.aggregate()__* to pass the data into the function *__JB_normality_test__*

> Aggregate using one or more operations over the specified axis.

Source: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.aggregate.html


In [14]:
Dist_Check["JB Normality Test"]=hfi.aggregate(JB_normality_test)
Dist_Check

Mean  Median  Mean < Median  Skewness  \
Convertible Arbitrage   0.005508  0.0065           True -2.639592   
CTA Global              0.004074  0.0014          False  0.173699   
Distressed Securities   0.006946  0.0089           True -1.300842   
Emerging Markets        0.006253  0.0096           True -1.167067   
Equity Market Neutral   0.004498  0.0051           True -2.124435   
Event Driven            0.006344  0.0084           True -1.409154   
Fixed Income Arbitrage  0.004365  0.0055           True -3.940320   
Global Macro            0.005403  0.0038          False  0.982922   
Long/Short Equity       0.006331  0.0079           True -0.390227   
Merger Arbitrage        0.005356  0.0060           True -1.320083   
Relative Value          0.005792  0.0067           True -1.815470   
Short Selling          -0.001701 -0.0053          False  0.767975   
Funds Of Funds          0.004262  0.0052           True -0.361783   

                        Excess Kurtosis  JB Normality Test  
Convertible Arbitrage         20.280834              False  
CTA Global                    -0.047040               True  
Distressed Securities          4.889983              False  
Emerging Markets               6.250788              False  
Equity Market Neutral         14.218555              False  
Event Driven                   5.035828              False  
Fixed Income Arbitrage        26.842199              False  
Global Macro                   2.741679              False  
Long/Short Equity              1.523893              False  
Merger Arbitrage               5.738950              False  
Relative Value                 9.121208              False  
Short Selling                  3.117772              False  
Funds Of Funds                 4.070153              False